src/firefly/js/visualize/ui/MouseReadout.jsx

```diff
@@ -77,25 +85,73 @@ MouseReadoutLock.propTypes = {
 
 
 
-export const DataReadoutItem= memo(({lArea, vArea, labelStyle={}, valueStyle={},
-                                        label='', value='', prefChangeFunc=undefined}) => {
+export const DataReadoutItem= memo(({lArea, vArea, cArea, labelStyle={}, valueStyle={}, showCopy=false,
+                                        label='', value='', copyValue='', prefChangeFunc=undefined}) => {
     const lS= lArea ? {gridArea:lArea,...labelStyle} : labelStyle;
     const vS= vArea ? {gridArea:vArea,...valueStyle} : valueStyle;
+    const cS= cArea ? {gridArea:cArea, overflow:'hidden', height:13} : undefined;
     const labelClass= prefChangeFunc ? 'mouseReadoutLabel mouseReadoutClickLabel' : 'mouseReadoutLabel';
+    const copyTitle= `Copy to clipboard: ${copyValue||value}`;
+
+    let clipComponent= undefined;
+    if (cArea) {
+        clipComponent= (value && showCopy) ?
+            <CopyToClipboard style={cS} title={copyTitle} value={copyValue||value} /> : <div style={cS}/>;
+    }
+
+
     return (
         <Fragment>
             <div className={labelClass} style={lS} onClick={prefChangeFunc}>{label}</div>
             <div style={vS}> {value} </div>
+            {clipComponent}
         </Fragment>
     );
 });
 
 DataReadoutItem.propTypes = {
-    lArea:          string,   //  label grid Area used with css grid-template-areas
-    vArea:          string,   //  value grid Area used with css grid-template-areas
+    lArea:          string,   // label grid Area used with css grid-template-areas
+    vArea:          string,   // value grid Area used with css grid-template-areas
+    cArea:          string,   // clipboard grid Area used with css grid-template-areas
+    showCopy:       bool,     // show the copy icon
     labelStyle:     object,
     valueStyle:     object,
     label:          string,
     value:          oneOfType([number,string]),
+    copyValue:      string,   // for copy to clipboard, if specified, us this value other use value
     prefChangeFunc: func,
 };
+
+
+
+const buttonStyle= {
+    borderRadius:2,
+    backgroundColor:'rgba(255,255,255,.9',
+    border:'solid transparent',
+    borderWidth: '0 0 1px 0'
+};
```

Comment:
You are already using `MouseReadout.css`, why not put styles in there?

Reply:
good sugestion

Reply:
oh, I can do that after all.  `ToolbarButton` does not take a class parameter. I could add it but it would just make it more messy.